vcd/resource_vcd_catalog_media_test.go

```diff
@@ -0,0 +1,129 @@
+package vcd
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/vmware/go-vcloud-director/govcd"
+	"testing"
+)
+
+var TestAccVcdCatalogMedia = "TestAccVcdCatalogMediaBasic"
+var TestAccVcdCatalogMediaDescription = "TestAccVcdCatalogMediaBasicDescription"
+
+func TestAccVcdCatalogMediaBasic(t *testing.T) {
+
+	var catalogItem govcd.CatalogItem
+	var params = StringMap{
+		"Org":              testConfig.VCD.Org,
+		"Catalog":          testConfig.VCD.Catalog.Name,
+		"CatalogMediaName": TestAccVcdCatalogMedia,
+		"Description":      TestAccVcdCatalogMediaDescription,
+		"MediaPath":        testConfig.Media.MediaPath,
+		"UploadPieceSize":  testConfig.Media.UploadPieceSize,
+		"UploadProgress":   testConfig.Media.UploadProgress,
+	}
+
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+	configText := templateFill(testAccCheckVcdCatalogMediaBasic, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCatalogMediaDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: configText,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVcdCatalogMediaExists("vcd_catalog_media."+TestAccVcdCatalogMedia, &catalogItem),
+					resource.TestCheckResourceAttr(
+						"vcd_catalog_media."+TestAccVcdCatalogMedia, "name", TestAccVcdCatalogMedia),
+					resource.TestCheckResourceAttr(
+						"vcd_catalog_media."+TestAccVcdCatalogMedia, "description", TestAccVcdCatalogMediaDescription),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckVcdCatalogMediaExists(mediaName string, catalogItem *govcd.CatalogItem) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		catalogMediaRs, ok := s.RootModule().Resources[mediaName]
+		if !ok {
+			return fmt.Errorf("not found: %s", mediaName)
+		}
+
+		if catalogMediaRs.Primary.ID == "" {
+			return fmt.Errorf("no catalog media ID is set")
```

Comment:
"no catalog media ID is set" ==> either "no catalog media ID was set" or "catalog media ID is not set"